### PR TITLE
Add adapter-outboxer

### PR DESCRIPTION
**Pull request by sonallux**
Adds the outbox pattern to the adapter.
- Add outboxer service to adapter database
- Insert event into outbox table instead of publishing to RabbitMQ
- Remove all RabbitMQ stuff from the adapter
- Adjust integration and system test because event delivery does take slightly longer now

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -55,6 +48,30 @@ services:
     environment:
       - POSTGRES_USER=adapterservice
       - POSTGRES_PASSWORD=admin
+    command:
+      - '-cwal_level=logical'
+      - '-cmax_wal_senders=1'
+      - '-cmax_replication_slots=1'
```
Comment (georg-schwarz) [inline docker-compose.yml:55]: What's that about?

File: adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/api/amqp/AmqpPublisher.java [hunk 2]
```diff
@@ -39,21 +39,7 @@ public void publishImportFailure(Long id, String errMsg) {
...
-          Thread.sleep(properties.getAmqp().getPublishBackoff());
-        } catch (InterruptedException interruptedException) {
-          Thread.currentThread().interrupt();
-          throw new RuntimeException(interruptedException);
-        }
-        log.warn("Message publish failed ({}). Retrying in {}", retries, properties.getAmqp().getPublishBackoff());
-      }
-    }
-    log.error("Sending message {} to topic {} failed", message.toString(), topic);
+    var event = new OutboxEvent(topic, message);
```
Comment (georg-schwarz) [inline adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/api/amqp/AmqpPublisher.java:42]: I assume that the calling class will manage the transaction then? Just curious...

File: docker-compose.yml (hunk 1 above)
Comment (sonallux) [reply docker-compose.yml:55]: Debezium uses [Postgres's logical decoding feature](https://www.postgresql.org/docs/current/logicaldecoding-explanation.html) to get the write-ahead log content. This enables and configures that feature.

File: adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/api/amqp/AmqpPublisher.java (hunk 2 above)
Comment (sonallux) [reply adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/api/amqp/AmqpPublisher.java:42]: No, transactions are handled by the `@Transactional` annotations in the `DatasourceManager`. For example here: https://github.com/jvalue/open-data-service/blob/1e670cdf2d36aae12273a2a965e0dd3e84edfe21/adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/DatasourceManager.java#L49-L56

File: docker-compose.yml (hunk 1 above)
Comment (georg-schwarz) [reply docker-compose.yml:55]: please add a comment for that! ;-)

Comment (sonallux) [reply docker-compose.yml:55]: Done!

**Review by georg-schwarz — DISMISSED** [review]
LGTM - @lunedis pls also have a look ;-)

**Comment by sonallux** [comment]
I have just reverted a change that I made because of testing purposes, The `adapter-db` is now no longer exposed by default.

**Review by lunedis — APPROVED** [review]
LGTM

**Review by georg-schwarz — APPROVED** [review]

**Pull request merged**